pyfftw/pyfftw.pyx

```diff
@@ -1082,14 +1087,29 @@ cdef class FFTW:
         # Set the timelimit
         set_timelimit_func(_planning_timelimit)
 
-        # Finally, construct the plan
-        self._plan = self._fftw_planner(
-            self._rank, <fftw_iodim *>self._dims,
-            self._howmany_rank, <fftw_iodim *>self._howmany_dims,
-            <void *>np.PyArray_DATA(self._input_array),
-            <void *>np.PyArray_DATA(self._output_array),
-            self._direction, self._flags)
-
+        # Finally, construct the plan, after acquiring the global planner lock
+        # (so that only one python thread can plan at a time, as the FFTW
+        # planning functions are not thread-safe)
+        
+        # no self-lookups allowed in nogil block, so must grab all these first
+        cdef void *plan
+        cdef fftw_generic_plan_guru fftw_planner = self._fftw_planner
+        cdef int rank = self._rank
+        cdef fftw_iodim *dims = <fftw_iodim *>self._dims
+        cdef int howmany_rank = self._howmany_rank
+        cdef fftw_iodim *howmany_dims = <fftw_iodim *>self._howmany_dims
+        cdef void *_in = <void *>np.PyArray_DATA(self._input_array)
+        cdef void *_out = <void *>np.PyArray_DATA(self._output_array)
+        cdef int sign = self._direction
+        cdef int c_flags = self._flags
+        
+        plan_lock.acquire()
+        with nogil:
+            plan = fftw_planner(rank, dims, howmany_rank, howmany_dims, 
+                _in, _out, sign, c_flags)
```

Comment:
It's a weak requirement, but can we make this more pep8y (line up the arguments, unless all on new line).